### PR TITLE
Fix strides format for data format in contrib.layers.separable convolution2d

### PR DESCRIPTION
**Pull request by jcastagneri**
Addresses feature request from issue #10432. Continuation of feature implementation--initial changes did not address the case where num_outputs = None, in which case strides and the input channels must be reformulated according to data format. The previous commit is  rmlarsen@d52e15a.

I created a previous PR #12120 for a similar issue that @rmlarsen fixed in the above commit, but the strides were still mismatched to data format--I thought it would be cleaner if I just made a new PR.

Edit--I thought it would be cleaner because my CLA would be recognized, but it still hasn't. See the comment below.

### CODE REVIEW

**Comment by jcastagneri** [comment]
I don't know why my CLA is failing... The commits list my CLA email: joe@ctxrobotics.com. When I click the redirect link from the CLA bot, it shows I am covered under a corporate CLA. As far as I am aware, I have taken the necessary steps to be certified under the CLA. Anybody know what I am missing here?

**Comment by jcastagneri** [comment]
@rmlarsen Is there any movement on this issue? It's really just a one line fix plus the test update, but until it is merged the channels first separable convolution from contrib is wrong.

**Comment by jhseu** [comment]
Keep in mind that we can't look at the code until the CLA is resolved. It looks like you have commits without e-mails in the list of commits. Perhaps resolve that?

**Comment by drpngx** [comment]
@jcastagneri any luck on the CLA? We'll have to close this PR otherwise. Thanks.

**Comment by sb2nov** [comment]
@jcastagneri It might be useful to try checking the CLA incognito to make sure it is validating the correct account.

**Comment by drpngx** [comment]
Ah, @jcastagneri you have signed via a company. You need to have the company on your public github profile for the CLA checking to work. Thanks.

**Comment by jcastagneri** [comment]
Thanks for the fix @drpngx. I added the company name, but the company is not actually on Github. Will this still block the CLA bot? I just added 'ctxrobotics' in the company section of my bio, as this is what I believe to be the company name registered to the CLA.

**Comment by martinwicke** [comment]
@willnorris for knowledge of how exactly company CLAs work.

I believe your company has to have the email on file with the CLA, and that email must be the same as the one used in the commits.

**Comment by willnorris** [comment]
@jcastagneri: it looks like you need to add your ctx robotics email [to your github account](https://help.github.com/articles/adding-an-email-address-to-your-github-account/).

**Comment by jcastagneri** [comment]
I have just added the email. Do I have to kickoff the CLA bot manually or will it check automatically?

**Comment by drpngx** [comment]
Great, I think it's ready for review now.

@rmlarsen it looks like this is something you might have some prior knowledge about?

File: tensorflow/contrib/layers/python/layers/layers.py [hunk 1]
```diff
@@ -2563,7 +2563,7 @@ def separable_convolution2d(
           regularizer=weights_regularizer,
           trainable=trainable,
           collections=weights_collections)
-      strides = [1, stride_h, stride_w, 1]
+      strides = [1, 1, stride_h, stride_w] if data_format.startswith('NC') else [1, stride_h, stride_w, 1]
```
Comment (sguada) [inline tensorflow/contrib/layers/python/layers/layers.py]: Can you add a test where this change is tested? i.e. when num_outputs = None but depth_multiplier is Not.

Comment (jcastagneri) [reply tensorflow/contrib/layers/python/layers/layers.py]: @sguada As far as I am aware, my test already does this check (`testConvNCHW` in `layers_test.py`), which is a part of this PR. Specifically, the test will assert correct output shape size for NCHW data input twice: Once with `num_outputs = None`, and with `num_outputs=5`, a non-None test value. Both tests are done with `depth_multiplier = 2`, which I believe satisfies the needed tests for this change. Let me know if I have missed something!

**Review by sguada — APPROVED** [review]
Yeah I overlooked that.

**Comment by martinwicke** [comment]
Jenkins. test this please.

**Pull request merged**